bit_manipulation/missing_number.py

```diff
@@ -11,11 +11,16 @@ def find_missing_number(nums: list[int]) -> int:
     Example:
         >>> find_missing_number([0, 1, 3, 4])
         2
+        >>> find_missing_number([1, 3, 4, 5, 6])
+        2
+        >>> find_missing_number([6, 5, 4, 2, 1])
+        3
     """
-    n = len(nums)
-    missing_number = n
+    low = min(nums[0], nums[-1])
+    high = max(nums[0], nums[-1])
```

Comment:
This wouldn't work if the numbers are out of order. Try adding this doctest and you'll see what I mean:
```py
>>> find_missing_number([6, 1, 5, 3, 4])
2
```

Reply:
Hi @tianyizheng02,
Thank you for your feedback. I have added the doctest you suggested, and the issue with out of order numbers has been resolved. Please review the changes at your convenience.